tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_Shape.cpp

```diff
@@ -0,0 +1,442 @@
+        sqrt_2_by_pi * std::real(gsl::at(gsl::at(coefs, l), 0));
+    for (size_t m = 1; m <= std::min(l, m_max); ++m) {
+      iter.set(l, m);
+      spherepack_coefs[iter()] =
+          pow(-1, m) * sqrt_2_by_pi * std::real(gsl::at(gsl::at(coefs, l), m));
+      iter.set(l, -m);
+      spherepack_coefs[iter()] =
+          pow(-1, m) * sqrt_2_by_pi * std::imag(gsl::at(gsl::at(coefs, l), m));
+    }
+  }
+  return spherepack_coefs;
+}
+
+// Generates the map, time, and a FunctionOfTime
+template <typename TransitionFunc>
+void generate_random_map_time_and_f_of_time(
+    const gsl::not_null<CoordinateMaps::TimeDependent::Shape<TransitionFunc>*>
+        map,
+    const gsl::not_null<double*> time,
+    const gsl::not_null<FunctionsOfTimeMap*> functions_of_time, size_t l_max,
+    size_t m_max, const std::array<double, 3>& center,
+    const TransitionFunc& transition_func, const DataVector& ylm_coefs,
+    gsl::not_null<std::mt19937*> generator) {
+  std::string f_of_t_name{"test_coefs"};
+
+  *map = CoordinateMaps::TimeDependent::Shape<TransitionFunc>(
+      center, l_max, m_max, transition_func, f_of_t_name);
+  // Choose a random time for evaluating the FunctionOfTime
+  std::uniform_real_distribution<double> time_dis{-1.0, 1.0};
+  *time = time_dis(*generator);
+  std::uniform_real_distribution<double> coef_dis{0.2, 2.};
+
+  // Set derivatives to zero
+  const auto dtcoefs = DataVector(ylm_coefs.size(), 0.);
+  const auto ddtcoefs = DataVector(ylm_coefs.size(), 0.);
+
+  const std::array<DataVector, 3> initial_coefficients = {ylm_coefs, dtcoefs,
+                                                          ddtcoefs};
+
+  std::uniform_real_distribution<> dt_dis{0.1, 0.5};
+  const double initial_time{*time - dt_dis(*generator)};
+  const double expiration_time{*time + dt_dis(*generator)};
+  (*functions_of_time)[f_of_t_name] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time, initial_coefficients, expiration_time);
+}
+
+template <typename TransitionFunc>
+void test_map_helpers(const TransitionFunc& transition_func, size_t l_max,
+                      size_t m_max, gsl::not_null<std::mt19937*> generator) {
+  std::uniform_real_distribution dist{-10., 10.};
+  const auto center =
+      make_with_random_values<std::array<double, 3>>(generator, dist, 3);
+
+  FunctionsOfTimeMap functions_of_time{};
+  double time{};
+  auto map = CoordinateMaps::TimeDependent::Shape<TransitionFunc>{};
+  const auto random_coefs = generate_random_coefs(l_max, m_max, generator);
+  const auto spherepack_coefs =
+      convert_coefs_to_spherepack(random_coefs, l_max, m_max);
+  generate_random_map_time_and_f_of_time(
+      make_not_null(&map), make_not_null(&time),
+      make_not_null(&functions_of_time), l_max, m_max, center, transition_func,
+      spherepack_coefs, generator);
+
+  const auto random_point =
+      make_with_random_values<std::array<double, 3>>(generator, dist, 3);
```

Comment:
With all the randomness going on, could you please run the test O(1e4) times to make sure edge cases work? `ctest --repeat-until-fail 10000 -L ...`